src/tls.c

```diff
@@ -2617,7 +2617,7 @@ static int TLSX_TCA_Parse(WOLFSSL* ssl, const byte* input, word16 length,
                     return BUFFER_ERROR;
                 ato16(input + offset, &idSz);
                 offset += OPAQUE16_LEN;
-                if (offset + idSz > length)
+                if (idSz > length - offset)
```

Comment:
What if `length < offset` is true? That will need to be checked as well or perhaps this function would be better having "offset" as word32?

Reply:
Good catch. Since length is also a word16, I just added a check for `(offset > length)`